### PR TITLE
Fix bug with non-unique IDs and names.

### PR DESCRIPTION
**Pull request by cahrens**
@andy-armstrong and @efischer19 this is ready for review. The sandbox is rubric.sandbox.edx.org (and it is configured for running the bok choy tests).

https://rubric.sandbox.edx.org/courses/course-v1:edx+ORA203+course/courseware/a4dfec19cf9b4a6fb5b18be6ccd9cecc/

Nothing visually should have changed, but the HTML structure has changed somewhat in the rubric template (to remove a list).

@cptvitamin If you could please review the HTML changes as well, that would be excellent. When this PR gets thumbs-ups, I will be merging to the branch you are currently reviewing.

TNL-3958

### CODE REVIEW

File: openassessment/templates/openassessmentblock/oa_rubric.html [hunk 1]
```diff
@@ -15,44 +15,46 @@ <h4 class="question__title ui-toggle-visibility__control">
...
-                                   value="{{ option.name }}"
-                                   aria-labelledby="{{ rubric_type }}__assessment__rubric__prompt--{{ criterion.order_num }}"/>
-                            <label for="{{ rubric_type }}__assessment__rubric__question--{{ criterion.order_num }}__{{ option.order_num }}"
-                                   class="answer__label"
-                                   >{{ option.label }}</label>
-                        </div>
-                        <div class="wrapper--metadata">
-                            <span class="answer__tip">{{ option.explanation }}</span>
-                            <span class="answer__points">{{ option.points }} <span class="answer__points__label">{% trans "points" %}</span></span>
+                <div class="question__answers">
```
Comment (cahrens) [inline openassessment/templates/openassessmentblock/oa_rubric.html]: @cptvitamin recommended removing ol/li structure here and using divs instead.

Also moved aria-labelledby to the group div (from individual inputs) and changed "name" attribute to "data-criterion-name", as name attribute should not have spaces in it (and criterion.name is specified by staff in Studio settings).

File: openassessment/xblock/static/js/spec/lms/oa_staff_area.js [hunk 2]
```diff
@@ -435,6 +435,7 @@ describe('OpenAssessment.StaffAreaView', function() {
 
         describe('Staff Grade Override', function() {
             var staffAreaTab = "staff-tools";
+            var staffType = "override";
```
Comment (cahrens) [inline openassessment/xblock/static/js/spec/lms/oa_staff_area.js]: Note to self: rename as gradeType.

File: test/acceptance/accessibility.py [hunk 3]
```diff
@@ -216,6 +216,35 @@ def test_training_peer_self_staff_override(self):
...
+    def setUp(self):
+        super(FullWorkflowRequiredA11yTest, self).setUp('full_workflow_staff_required', staff=True)
+        self.staff_area_page = StaffAreaPage(self.browser, self.problem_loc)
+
+    def test_multiple_rubrics(self):
+        """
+        Test accessibility when both the staff override and the full staff grading
+        rubric forms have been opened.
+        """
+        # Create a learner with submission, training, and self assessment completed.
```
Comment (cahrens) [inline test/acceptance/accessibility.py]: This test caught the duplicate IDs and the non-unique name values.

File: test/acceptance/tests.py [hunk 4]
```diff
@@ -829,16 +858,6 @@ def peer_grade_exists():
             "Learner still not graded after {} additional attempts".format(max_attempts)
         )
 
-
-class FullWorkflowBaseTest(OpenAssessmentTest, FullWorkflowMixin):
```
Comment (cahrens) [inline test/acceptance/tests.py]: To allow reuse with the accessibility test, I got rid of FullWorkflowBaseTest and moved the methods directly into FullWorkflowMixin.

File: openassessment/xblock/static/js/src/lms/oa_rubric.js [hunk 5]
```diff
@@ -33,12 +33,13 @@ OpenAssessment.Rubric.prototype = {
         var feedback = {};
         $(selector, this.element).each(
             function(index, sel) {
+                var criterionName = $(sel).data('criterion-name');
```
Comment (cahrens) [inline openassessment/xblock/static/js/src/lms/oa_rubric.js]: Note to self: create a helper method.

File: openassessment/templates/openassessmentblock/oa_rubric.html [hunk 6]
```diff
@@ -15,44 +15,47 @@ <h4 class="question__title ui-toggle-visibility__control">
...
+                    <div role="group" aria-labelledby="{{ rubric_type }}__assessment__rubric__prompt--{{ criterion.order_num }}">
+                        {% for option in criterion.options %}
+                        <div class="answer">
+                            <div class="wrapper--input">
+                                <input type="radio"
+                                       data-criterion-name="{{ criterion.name }}"
+                                       id="{{ rubric_type }}__assessment__rubric__question--{{ criterion.order_num }}__{{ option.order_num }}"
+                                       class="answer__value"
+                                       value="{{ option.name }}"
+                                       name="{{ rubric_type }}__assessment__rubric__question--{{ criterion.order_num }}"/>
```
Comment (cahrens) [inline openassessment/templates/openassessmentblock/oa_rubric.html]: @cptvitamin it turns out that we do need name. It is what makes radio buttons groups single-selection only.

File: openassessment/templates/openassessmentblock/oa_rubric.html [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {% spaceless %}
 {% load i18n %}
-<fieldset class="assessment__fields">
+<div class="assessment__fields">
```
Comment (cahrens) [inline openassessment/templates/openassessmentblock/oa_rubric.html]: @cptvitamin said this use of fieldset is just wrong (legend must appear immediately after fieldset if it is used).

**Comment by andy-armstrong** [comment]
:+1: Great clean up!

**Comment by efischer19** [comment]
:+1: Looks great!

**Pull request merged**